client/src/components/home/modules-section.tsx

```diff
@@ -0,0 +1,335 @@
+          molestie vel. Ullamcorper risus fermentum curabitur consequat arcu convallis. Eget velit
+          purus auctor adipiscing nulla lacinia semper hendrerit sociis.
+        </div>
+      </>
+    ),
+    modalList: [
+      'Lorem Ipsum ...',
+      'Lorem Ipsum ...',
+      'Lorem Ipsum ...',
+      'Lorem Ipsum ...',
+      'Lorem Ipsum ...',
+    ],
+  },
+  datasets: {
+    tooltipText:
+      'Lorem ipsum dolor sit amet consectetur. Sit blandit a mauris varius id. In leo in adipiscing in. Aliquet fringilla tortor faucibus elementum. Iaculis consequat et turpis ut imperdiet arcu pellentesque.',
+    modalText: (
+      <>
+        <div className="text-base">
+          Lorem ipsum dolor sit amet consectetur. Sit blandit a mauris varius id. In leo in
+          adipiscing in. Aliquet fringilla tortor faucibus elementum. Iaculis consequat et turpis ut
+          imperdiet arcu pellentesque.
+        </div>
+        <div className="text-sm">
+          Lorem ipsum dolor sit amet consectetur. Proin id massa consequat cursus nulla pharetra.
+          Adipiscing non lobortis risus quam feugiat feugiat. Suspendisse proin donec turpis augue
+          non congue neque nec amet. Interdum blandit lacus vivamus amet cursus. Eget nisl est
+          feugiat sit. Leo a lectus sagittis id dolor amet vitae orci massa. Nunc id congue lacus
+          molestie vel. Ullamcorper risus fermentum curabitur consequat arcu convallis. Eget velit
+          purus auctor adipiscing nulla lacinia semper hendrerit sociis.
+        </div>
+      </>
+    ),
+    modalList: [
+      'Lorem Ipsum ...',
+      'Lorem Ipsum ...',
+      'Lorem Ipsum ...',
+      'Lorem Ipsum ...',
+      'Lorem Ipsum ...',
+    ],
+  },
+};
+
+type Slug = keyof typeof modulesInfo;
+
+type HoverTooltipProps = {
+  children: React.ReactNode;
+  tooltipOpen: boolean;
+  setTooltipOpen: (open: boolean) => void;
+  intro: React.ReactNode;
+  slug: Slug;
+  colorClass: string;
+};
+
+const HoverTooltip = ({
+  children,
+  tooltipOpen,
+  setTooltipOpen,
+  intro,
+  slug,
+  colorClass,
+}: HoverTooltipProps) => {
+  return (
+    <div onMouseEnter={() => setTooltipOpen(true)} onMouseLeave={() => setTooltipOpen(false)}>
+      <Tooltip open={tooltipOpen}>
+        <TooltipTrigger asChild>{children}</TooltipTrigger>
+        <TooltipPortal>
+          <TooltipContent
+            className={cn(
+              'absolute -left-[162.5px] -top-[63px] z-50 flex h-[292px] w-[325px]  cursor-pointer flex-col gap-4 p-6',
+              colorClass,
+            )}
+          >
+            {intro}
+            <div className=" text-sm leading-normal text-gray-700">
+              {modulesInfo[slug].tooltipText}
+            </div>
+            <div className="mt-auto flex items-center justify-end">
+              Learn more
+              <ArrowRight className="ml-6" />
+            </div>
+          </TooltipContent>
+        </TooltipPortal>
+      </Tooltip>
+    </div>
```

Comment:
I think you overcomplicated the data management. The parent component doesn't have to own the tooltip's state.

Why didn't you do this directly?
```suggestion
      <Tooltip>
        <TooltipTrigger asChild>{children}</TooltipTrigger>
        <TooltipPortal>
          <TooltipContent
            className={cn(
              'absolute -left-[162.5px] -top-[63px] z-50 flex h-[292px] w-[325px]  cursor-pointer flex-col gap-4 p-6',
              colorClass,
            )}
          >
            {intro}
            <div className=" text-sm leading-normal text-gray-700">
              {modulesInfo[slug].tooltipText}
            </div>
            <div className="mt-auto flex items-center justify-end">
              Learn more
              <ArrowRight className="ml-6" />
            </div>
          </TooltipContent>
        </TooltipPortal>
      </Tooltip>
```

Reply:
There was a problem in this case. The tooltip was intermittently opening and closing because it couln't decide if the focus was on the trigger or the content. I think this tooltip is not actually made for this case in which the content is over the trigger. So I have to create a parent component that group both of them to be the source for changes

Reply:
Look at this suggestion. I think it addresses the problem you're referring to.
[suggestion.zip](https://github.com/Orcasa-Platform/orcasa/files/13392233/suggestion.zip)